### ISSUE TITLE
libdexhelper.so issue

### ISSUE THREAD

**Issue by oppressor1761** [GrapheneOS/os_issue_tracker#3917]
An [app](https://dynamic.12306.cn/otn/appdownload/12306v5.8.2.12.apk) crashes after its latest update.
```
type: crash
osVersion: google/husky/husky:14/AP2A.240805.005/2024082200:user/release-keys
package: com.MobileTicket:271
process: com.MobileTicket
processUptime: 0 + 0 ms
installer: com.android.packageinstaller

signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0001d77eb3fa4001

backtrace:
      #00 pc 000000000003594c  /data/app/~~9X0zxU3_VQjuespDrN4XJA==/com.MobileTicket-9BLweKHXPIW7Aw4g0wQktw==/lib/arm64/libDexHelper.so


```
This looks like the issue  #3317.  It is a different app though. I'm not sure what's the behavior on the stock OS.

**Comment by muhomorr** [comment]
How to reproduce this crash?

**Comment by muhomorr** [comment]
@oppressor1761 Is this app installed in a secondary user? This app launches fine on my device without Exploit protection compatibility mode in the main user, but crashes in a secondary user.

**Comment by mochaaP** [comment]
I think this is due to an ART change: https://android.googlesource.com/platform/art/+/c7f5eb7654a9689f03dc0f038ad802758ed24623

**Comment by szescxz** [comment]
While I'm not a GrapheneOS user, I managed to reproduce the crash after applying one of the August Google Play system updates on stock Samsung One UI 6.1, while staying on July security patch (no OS updates).
The problematic Google Play system update caused an app optimization on first boot. Before the update I was able to launch the app normally.

I believe this issue affects all apps protected with `libDexHelper.so` by [Bangbang Security](https://dev.bangcle.com/home/introduction).
